pkg/storage/factory_test.go

```diff
@@ -45,6 +50,60 @@ func TestFactoryStop(t *testing.T) {
 	store.Stop()
 }
 
+func TestNamedStores(t *testing.T) {
+	tempDir := t.TempDir()
+
+	// config for BoltDB Shipper
+	boltdbShipperConfig := shipper.Config{}
+	flagext.DefaultValues(&boltdbShipperConfig)
+	boltdbShipperConfig.ActiveIndexDirectory = path.Join(tempDir, "index")
+	boltdbShipperConfig.SharedStoreType = "filesystem.named-store"
+	boltdbShipperConfig.CacheLocation = path.Join(tempDir, "boltdb-shipper-cache")
+	boltdbShipperConfig.Mode = indexshipper.ModeReadWrite
+
+	cfg := Config{
+		NamedStores: NamedStores{
+			Filesystem: map[string]local.FSConfig{
+				"named-store": {Directory: path.Join(tempDir, "chunks")},
+			},
+		},
+		BoltDBShipperConfig: boltdbShipperConfig,
+	}
+
+	schemaConfig := config.SchemaConfig{
+		Configs: []config.PeriodConfig{
+			{
+				From:       config.DayTime{Time: timeToModelTime(parseDate("2019-01-01"))},
+				IndexType:  "boltdb-shipper",
+				ObjectType: "filesystem.named-store",
+				Schema:     "v9",
+				IndexTables: config.PeriodicTableConfig{
+					Prefix: "index_",
+					Period: time.Hour * 168,
+				},
+			},
+		},
+	}
+
+	limits, err := validation.NewOverrides(validation.Limits{}, nil)
+	require.NoError(t, err)
+
+	t.Run("period config referring to configured named store", func(t *testing.T) {
+		store, err := NewStore(cfg, config.ChunkStoreConfig{}, schemaConfig, limits, cm, nil, util_log.Logger)
+		require.NoError(t, err)
```

Comment:
I'm not sure just testing for the absense of an error is sufficient here. I think we should be validating that the expected config was selected.

Reply:
cannot access the internals of store as it is private to a different pkg. And I feel mocking might not be an option here since the client init happens a few layers below.

I am thinking of adding an integration test: push a few log lines and ensure that the chunks and index are stored in the dir configured for the filesystem named store. wdyt?